src/main/java/mods/railcraft/common/blocks/single/TileTradeStation.java

```diff
@@ -125,10 +122,10 @@ private boolean attemptTrade(List<EntityVillager> villagers, int tradeSet) {
                 if (recipe.isRecipeDisabled())
                     continue;
                 //noinspection ConstantConditions
-                if (recipe.getItemToBuy() != null && !InvTools.isItemLessThanOrEqualTo(recipe.getItemToBuy(), buy1))
+                if (recipe.getItemToBuy() != InvTools.emptyStack() && !InvTools.isItemLessThanOrEqualTo(recipe.getItemToBuy(), buy1))
```

Comment:
This probably isn't 100% safe, use isEmpty(stack) in all these tests instead.

Reply:
Yeah made that fault.